java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -1011,4 +1013,92 @@ public void zcard(BaseClient client) {
                 assertThrows(ExecutionException.class, () -> client.zcard("foo").get());
         assertTrue(executionException.getCause() instanceof RequestException);
     }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void xadd(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        String field1 = UUID.randomUUID().toString();
+        String field2 = UUID.randomUUID().toString();
+
+        assertNull(
+                client
+                        .xadd(
+                                key,
+                                Map.of(field1, "foo0", field2, "bar0"),
+                                StreamAddOptions.builder().makeStream(Boolean.FALSE).build())
+                        .get());
+
+        String timestamp1 = "0-1";
+        assertEquals(
+                timestamp1,
+                client
+                        .xadd(
+                                key,
+                                Map.of(field1, "foo1", field2, "bar1"),
+                                StreamAddOptions.builder().id(timestamp1).build())
+                        .get());
+
+        assertNotNull(client.xadd(key, Map.of(field1, "foo2", field2, "bar2")).get());
+        if (client instanceof RedisClient) {
+            assertEquals(2L, ((RedisClient) client).customCommand(new String[] {"XLEN", key}).get());
```

Comment:
Add a TODO

Reply:
introduce a helper function since you repeat this

Reply:
I thinking that... but it will get removed when we implement xlen

Reply:
true